src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -0,0 +1,499 @@
+                    var additionalDocuments = CreateDocumentInfos(projectFileInfo.AdditionalDocuments, projectId, commandLineArgs.Encoding);
+                    CheckForDuplicateDocuments(documents, additionalDocuments, projectPath, projectId);
+
+                    var analyzerReferences = ResolveAnalyzerReferences(commandLineArgs);
+
+                    var resolvedReferences = await ResolveReferencesAsync(projectId, projectFileInfo, commandLineArgs, cancellationToken).ConfigureAwait(false);
+
+                    return ProjectInfo.Create(
+                        projectId,
+                        version,
+                        projectName,
+                        assemblyName,
+                        language,
+                        projectPath,
+                        outputFilePath: projectFileInfo.OutputFilePath,
+                        outputRefFilePath: projectFileInfo.OutputRefFilePath,
+                        compilationOptions: compilationOptions,
+                        parseOptions: parseOptions,
+                        documents: documents,
+                        projectReferences: resolvedReferences.ProjectReferences,
+                        metadataReferences: resolvedReferences.MetadataReferences,
+                        analyzerReferences: analyzerReferences,
+                        additionalDocuments: additionalDocuments,
+                        isSubmission: false,
+                        hostObjectType: null);
+                });
+            }
+
+            private static VersionStamp GetProjectVersion(string projectFilePath)
+            {
+                if (!string.IsNullOrEmpty(projectFilePath) && File.Exists(projectFilePath))
+                {
+                    return VersionStamp.Create(File.GetLastWriteTimeUtc(projectFilePath));
+                }
+                else
+                {
+                    return VersionStamp.Create();
+                }
+            }
+
+            private static string GetAssemblyNameFromProjectPath(string projectFilePath)
+            {
+                var assemblyName = Path.GetFileNameWithoutExtension(projectFilePath);
+
+                // if this is still unreasonable, use a fixed name.
+                if (string.IsNullOrWhiteSpace(assemblyName))
+                {
+                    assemblyName = "assembly";
+                }
+
+                return assemblyName;
+            }
+
+            private IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(CommandLineArguments commandLineArgs)
+            {
+                var analyzerService = GetWorkspaceService<IAnalyzerService>();
+                var analyzerLoader = analyzerService.GetLoader();
+
+                foreach (var path in commandLineArgs.AnalyzerReferences.Select(r => r.FilePath))
+                {
+                    var fullPath = Path.GetFullPath(path);
+                    if (File.Exists(fullPath))
```

Comment:
some uses FileUtilties stuff, and some not?

Reply:
I believe much of this existed before FileUtilities but later fixes used it. I'll do sweep to make that consistent.

Reply:
Note that there isn't a `FileUtilities.Exists(...)` method. Looking around Roslyn.sln it appears that we use `File.Exists(...)` all over the place.